api/controllers/tags.js

```diff
@@ -0,0 +1,28 @@
+const { Tag } = require("../models/tag.js");
+
+async function getAllTags() {
+  return Tag.find({});
+}
+
+async function addTag(data) {
+  const tag = new Tag({
+    tagName: data.tagName,
+    post: [],
```

Comment:
What about creating tag with existing posts? Keep it as default for optional value`posts: req.body.posts || []`

Reply:
Fixed